src/org/traccar/protocol/GranitProtocolEncoder.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright 2016 Anton Tananaev (anton.tananaev@gmail.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+ package org.traccar.protocol;
+
+import java.nio.ByteOrder;
+import java.nio.charset.StandardCharsets;
+
+import org.jboss.netty.buffer.ChannelBuffer;
+import org.jboss.netty.buffer.ChannelBuffers;
+import org.traccar.BaseProtocolEncoder;
+import org.traccar.helper.Log;
+import org.traccar.model.Command;
+
+public class GranitProtocolEncoder extends BaseProtocolEncoder {
+    @Override
+    protected Object encodeCommand(Command command) {
+
+        ChannelBuffer commandString;
+
+        switch (command.getType()) {
+        case Command.TYPE_IDENTIFICATION:
+            commandString = ChannelBuffers.directBuffer(ByteOrder.LITTLE_ENDIAN, 12);
```

Comment:
Less hardcoded values is better. For dynamic buffer you can use length 0 (EMPTY_BUFFER) and it will automatically resize to needed size when you add data.